frame/evm-system/src/lib.rs

```diff
@@ -160,16 +160,16 @@ impl<T: Config> Pallet<T> {
 
 	/// Increment a particular account's nonce by 1.
 	pub fn inc_account_nonce(who: &<T as Config>::AccountId) {
+		let was_existed = Self::account_exists(who);
```

Comment:
`was_existed` is not a valid tenses mix in english

Reply:
What about this taking into account https://github.com/humanode-network/frontier/pull/151#pullrequestreview-2588106250 - https://github.com/humanode-network/frontier/pull/151/commits/4366fb23597608d1b97d1c3ebc1a7819716fd71f ?

Reply:
No, now it is just wrong; had to look up where `account_exist` comes from to figure out what's really going on here any the confusion.

The idea is you'd do something like this:

```shell
let existed_before = Account::<T>::mutate(who, |a| {
    let existed_before = a == &Default::default();
    a += <T as Config>::Index::one();
    existed_before
});
if !existed_before {
    Self::on_created_account(who.clone());
}
```

Reply:
I really think we need to make some means of access for this `Account` storage value of ours; for instance, a `try_mutate_account` and `mutate_account` fns that would provide the following-ish logic:

```rust
let (existed_before, result) = Account::<T>::mutate(who, |a| {
    let existed_before = a == &Default::default();
    let result = f(a);
    (existed_before, result)
});
if !existed_before {
    Self::on_created_account(who.clone());
}
result
```

Reply:
ref: https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/system/src/lib.rs#L1754

Reply:
Actually, maybe just implement `StoredMap` and use it for `inc_nonce`

Reply:
At the very least, let's explain why we shouldn't just use a `StoredMap::try_mutate_exists` here... Or just use it :D

Reply:
Sure, we can, and I'm really would like to do it. But I suggest to use just `mutate_exists` as we are always return Ok - https://github.com/humanode-network/frontier/pull/151/commits/1235fb2abccb62b4acdc9dca62bea5831b731129

Reply:
That `mutate_exists` - is it from `StoredMap` or `StorageMap`?

Reply:
Also, if we use `StorageMap::mutate_exists` - then we don't need to check for `== Default...` as we already know whether the value was absent or not.

Reply:
I guess I'm confused now as it was so simple at the beginning :) Can't get your point now

Reply:
This looks fine though, but I really want to focus our attention on the fact that we have a perfectly fine `impl StoredMap for Pallet` that can send the `on_account_created`...

Reply:
There a lot of different options here in general starting from simple check if it was existed before by using in a different way `try_mutate_exists` and `mutate_exists`...

Reply:
All I'm saying is look at what  *out implementation of* `StoredMap` does. If we can use that it would be kind of nice, don't you think? As then we only need to define the logic of emitting the `on_created_account` in one place...

Reply:
Now I got you point, good suggestion, let me see on it

Reply:
Hmm... it would be kind of nice, but we already have implementation of `StoredMap<<T as Config>::AccountId, <T as Config>::AccountData>` that doesn't allow to operate with nonce. It's not allowed to implement additional StoredMap due to having conflict :(

In case extending current implementation with using nonce - then we need to refactor `pallet-evm-balances` in a bit

Reply:
> Hmm... it would be kind of nice, but we already have implementation of `StoredMap<<T as Config>::AccountId, <T as Config>::AccountData>` that doesn't allow to operate with nonce. It's not allowed to implement additional StoredMap due to having conflict :(

Yes, my thinking as well. Alright then.



> In case extending current implementation with using nonce - then we need to refactor `pallet-evm-balances` in a bit

We won't be able to, since this is the API that ties together balances and accounts implementation. Not under our control.

Reply:
What we have in this PR currently is good enough.